log10/completions/completions.py

```diff
@@ -0,0 +1,223 @@
+import json
+
+import click
+import httpx
+import rich
+import tqdm
+from rich.console import Console
+from rich.table import Table
+
+from log10._httpx_utils import _get_time_diff, _try_get
+from log10.llm import Log10Config
+
+
+# TODO: Support data filters
+_log10_config = Log10Config()
+
+
+def _get_completion(id: str) -> httpx.Response:
+    url = f"{_log10_config.url}/api/completions/{id}?organization_id={_log10_config.org_id}"
+    return _try_get(url)
+
+
+def _get_tag_id(tag: str) -> str:
+    url = f"{_log10_config.url}/api/tags/search?organization_id={_log10_config.org_id}&query={tag}"
+    res = _try_get(url)
+    data = res.json()["data"]
+    if not data:
+        return []
+
+    if len(data) == 1:
+        return data[0]["id"]
+    else:
+        for item in data:
+            if item["name"] == tag:
+                return item["id"]
+        return []
+
+
+def _get_tag_ids(tags):
+    tag_ids = []
+    for tag in tags.split(","):
+        tag_id = _get_tag_id(tag)
+        if tag_id:
+            tag_ids.append(tag_id)
+    tag_ids_str = ",".join(tag_ids)
+    return tag_ids_str
+
+
+def _get_completions_url(limit, offset, tags, from_date, to_date, base_url, org_id, printout=True):
+    tag_ids_str = _get_tag_ids(tags) if tags else ""
+    if tag_ids_str and printout:
+        rich.print(f"Filter with tags: {tags}")
+
+    date_range = _get_valid_date_range(from_date, to_date)
+    if date_range and printout:
+        rich.print(f"Filter with created date: {date_range['from'][:10]} to {date_range['to'][:10]}")
+
+    url = f"{base_url}/api/completions?organization_id={org_id}&offset={offset}&limit={limit}&tagFilter={tag_ids_str}&createdFilter={json.dumps(date_range)}&sort=created_at&desc=true&ids="
+    return url
+
+
+def _get_valid_date_range(from_date, to_date):
+    if not from_date and not to_date:
+        return {}
+
+    if from_date is None or to_date is None:
+        raise click.UsageError("Both --from and --to must be set together.")
+
+    if from_date >= to_date:
+        raise click.UsageError(f"from_date {from_date} must be earlier than to_date {to_date}")
+
+    parsed_from_date = from_date.replace(hour=8).strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3] + "Z"
+    parsed_to_date = to_date.replace(hour=8).strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3] + "Z"
+
+    date_range = {"from": parsed_from_date, "to": parsed_to_date}
+    return date_range
+
+
+def _render_completions_table(completions_data, total_completions):
+    data_for_table = []
+    for completion in completions_data:
+        if completion["kind"] == "completion":
+            prompt = completion["request"]["prompt"] if completion.get("request", {}) else ""
```

Comment:
You may want to guard some of these accesses, in case request, response etc aren't available - maybe `.get("<field>", {})` throughout?